File: API/src/main/java/com/mareenraj/api/entity/Employee.java
package com.mareenraj.api.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(
        name = "employees",
        indexes = {
                @Index(name = "idx_employee_email", columnList = "email")
        }
)
@Getter
@Setter
@NoArgsConstructor
public class Employee {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Long id;

    @Column(nullable = false)
    private String firstName;

    @Column(nullable = false)
    private String lastName;

    @Column(nullable = false)
    private String address;

    @Column(nullable = false, unique = true)
    private String email;

    @Column(nullable = false)
    private LocalDate joinDate;

    @Column(nullable = false)
    private double currentSalary;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private EmployeeStatus employeeStatus;

    @CreationTimestamp
    private LocalDateTime createdAt;

    @UpdateTimestamp
    private LocalDateTime updatedAt;
}

File: frontend/src/types/employee.ts
export interface EmployeeResponse {
  id: number;
  firstName: string;
  lastName: string;
  address: string;
  email: string;
  joinDate: string;
  currentSalary: number;
  employeeStatus: string;
}

export interface CreateEmployeeRequest {
  firstName: string;
  lastName: string;
  address: string;
  email: string;
  joinDate: string;
  currentSalary: number;
  employeeStatus: string;
}

export interface UpdateEmployeeRequest {
  firstName: string;
  lastName: string;
  address: string;
  email: string;
  joinDate: string;
  currentSalary: number;
  employeeStatus: string;
}
